src/interface.c

```diff
@@ -5156,6 +5158,62 @@ int sha1s_parse_hash (u8 *input_buf, u32 input_len, hash_t *hash_buf, MAYBE_UNUS
   return (PARSER_OK);
 }
 
+int totp_parse_hash (u8 *input_buf, u32 input_len, hash_t *hash_buf, MAYBE_UNUSED hashconfig_t *hashconfig)
+{
+  // this is going to start off like HMAC-SHA1
+  u32 *digest = (u32 *) hash_buf->digest;
+
+  salt_t *salt = hash_buf->salt;
+
+  token_t token;
+
+  token.token_cnt  = 2;
+
+  token.sep[0]     = hashconfig->separator;
+  token.len_min[0] = 6;
+  token.len_max[0] = 6;
+  token.attr[0]    = TOKEN_ATTR_VERIFY_LENGTH
+                   | TOKEN_ATTR_VERIFY_HEX;
+
+  token.len_min[1] = SALT_MIN;
+  token.len_max[1] = SALT_MAX;
+  token.attr[1]    = TOKEN_ATTR_VERIFY_LENGTH;
+
+  if (hashconfig->opts_type & OPTS_TYPE_ST_HEX)
+  {
+    token.len_min[1] *= 2;
+    token.len_max[1] *= 2;
+
+    token.attr[1] |= TOKEN_ATTR_VERIFY_DIGIT;
+  }
+
+  const int rc_tokenizer = input_tokenizer (input_buf, input_len, &token);
+
+  // now we need to reduce our hash into a token
+  int otp_code = hc_strtoul ((const char *) input_buf, NULL, 10);
+
+  if (rc_tokenizer != PARSER_OK) return (rc_tokenizer);
+
+  digest[1] = otp_code;
+
+  u8 *salt_pos = token.buf[1];
+
+  // convert ascii timestamp to ulong timestamp
+  u64 timestamp = hc_strtoul ((const char *) salt_pos, NULL, 10);
```

Comment:
there is a bug here, must be:
```
   u64 timestamp = hc_strtoull ((const char *) salt_pos, NULL, 10);
```
(i.e. use hc_strtoull () for 64 bit, instead of hc_strtoul () which would only allow 32 bit)